src/Quartz/Impl/StdSchedulerFactory.cs

```diff
@@ -508,11 +516,14 @@ protected virtual void ValidateConfiguration()
 
             Type tpType = loadHelper.LoadType(cfg.GetStringProperty(PropertyThreadPoolType)) ?? typeof(SimpleThreadPool);
 
+#if WINDOWS_THREADPOOL  // TODO (NetCore Port) - We can't use the ClrThreadPool on .NET Core because cross-plat threadpool implementations don't expose all the APIs we need
+                        // Use the SimpleThreadPool there for now and in the future create a new async-friendly threadpool that will work cross-plat.
             if (tpType == typeof(SimpleThreadPool))
```

Comment:
I'd like to just point out the current situation, we had SimpleThreadPool that used to handle fixed size worker pool: https://github.com/quartznet/quartznet/blob/quartznet-3/src/Quartz/Simpl/SimpleThreadPool.cs . This was the guarantee to have predictability over .NET own thread pool. Fixed size, fixed availability consumed only by Quartz workers.

Initial thought (before .NET Core portability talk) was to try to utilize TPL's example QueuedTaskScheduler (Masstransit for one uses this now in async world): https://github.com/quartznet/quartznet/blob/quartznet-3/src/Quartz/Util/QueuedTaskScheduler.cs

The current ClrThreadPool was just initial torso there as a TODO.

Thank you both for helping to find a solution. Your help is much appreciated as I'm very new to .NET Core limitations and possibilities.